### PR TITLE
[CI][macOS runners] Upload JUnit and Coverage to Datadog and Codecov

### PR DESCRIPTION
**Pull request by amenasria**
### What does this PR do?

This PR uploads:
- the JUnit data to Datadog to display the Unit Tests info in Test Visibility.
- the coverage data to Codecov.

### Motivation

Better visibility & coverage.

### Additional Notes

https://datadoghq.atlassian.net/browse/ACIX-421
https://datadoghq.atlassian.net/browse/ACIX-422

### CODE REVIEW

File: .gitlab/source_test/macos.yml [hunk 1]
```diff
@@ -117,7 +137,13 @@ lint_macos_gitlab_arm64:
 tests_macos_gitlab_amd64:
   extends: .tests_macos_gitlab
   tags: ["macos:monterey-amd64", "specific:true"]
+  after_script:
+    - !reference [.upload_junit_source]
+    - !reference [.upload_coverage]
```
Comment (CelianR) [inline .gitlab/source_test/macos.yml:142]: **❓ question:** ‏Would it be better to add another reference that combines both of them? It seems that they are always used together 🤷 (Something like `.upload_reports`?)

**Review by CelianR — APPROVED** [review]

File: .gitlab/source_test/macos.yml (hunk 1 above)
Comment (amenasria) [reply .gitlab/source_test/macos.yml:142]: We will combine them in the end, but for now it's still not used together everywhere (for arm jobs for example)

**Pull request merged**